### DISCUSSION TITLE
Set default value while XML Deserialization C#

### DISCUSSION THREAD

**Discussion by true-craft-super** [community/community#58121] [Programming Help]
### Body

Consider the below class

    public class Person
    {
        public string Name { get; set; }
        public int Age { get; set; }
    }

Serialized this Person class (Name & Age) with some value in XML file and deserialized return back. I want to assign some default value when Name value is Null or Empty when deserializing. I tried like below, but its not working.

    public class Person
    {
        private string _name;
    
        public string Name
        {
            get { return _name; }
            set
            {
                _name = value;
                if (_name == null)
                {
                    _name = "Some Name";
                }
            }
        }
    }

How to set default value for string and int field when deserializing the C# object using XmlSerializer.

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by dreamcatcher1020** [comment]
You have to initial property inside Contructor like [Own Pauling](https://stackoverflow.com/a/59389773/10649529)

Or you can auto-initialize like this:

    public class Person{
    public string Name { get; set; } = "Some name"
    public int Age { get; set; } = 18
    }
On your code demo.

It just working when you set something (anything, even null) to the property before Deserialization.
